main.tf

```diff
@@ -34,10 +34,13 @@ resource "azurerm_kubernetes_cluster" "main" {
     client_secret = var.client_secret
   }
 
-  addon_profile {
-    oms_agent {
-      enabled                    = true
-      log_analytics_workspace_id = azurerm_log_analytics_workspace.main.id
+  dynamic addon_profile {
+    for_each = var.enable_log_analytics_workspace ? ["log_analytics"] : []
+    content {
+      oms_agent {
+        enabled                    = true
+        log_analytics_workspace_id = azurerm_log_analytics_workspace.main[0].id
```

Comment:
What if the count is zero? Will there be any friendly error message / default behaviors, or anything will crash?

Reply:
In this case, it will not be zero. I could add something conditional equation. But the field can not be assigned with empty string anyway. I have test them in two conditions in fixture.